From 8b41a9fb9163c946da55b0a93c794ffffa32d23e Mon Sep 17 00:00:00 2001
From: AdrianDC <radian.dc@gmail.com>
Date: Wed, 18 Nov 2015 19:15:29 +0100
Subject: [PATCH 050/296] MountService: Prevent NPE with DropBoxManager

 * "java.lang.NullPointerException: Attempt to invoke virtual
    method 'void android.os.DropBoxManager.addText(java.lang.String,
    java.lang.String)' on a null object reference"
---
 .../core/java/com/android/server/StorageManagerService.java    | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/StorageManagerService.java b/services/core/java/com/android/server/StorageManagerService.java
index 70947a37238..8c93f372e09 100644
--- a/services/core/java/com/android/server/StorageManagerService.java
+++ b/services/core/java/com/android/server/StorageManagerService.java
@@ -1218,8 +1218,10 @@ class StorageManagerService extends IStorageManager.Stub
                 final long destroy = Long.parseLong(cooked[6]);
 
                 final DropBoxManager dropBox = mContext.getSystemService(DropBoxManager.class);
-                dropBox.addText(TAG_STORAGE_BENCHMARK, scrubPath(path)
-                        + " " + ident + " " + create + " " + run + " " + destroy);
+                if (dropBox != null) {
+                    dropBox.addText(TAG_STORAGE_BENCHMARK, scrubPath(path)
+                            + " " + ident + " " + create + " " + run + " " + destroy);
+                }
 
                 final VolumeRecord rec = findRecordForPath(path);
                 if (rec != null) {
@@ -1236,8 +1238,10 @@ class StorageManagerService extends IStorageManager.Stub
                 final long time = Long.parseLong(cooked[3]);
 
                 final DropBoxManager dropBox = mContext.getSystemService(DropBoxManager.class);
-                dropBox.addText(TAG_STORAGE_TRIM, scrubPath(path)
+                if (dropBox != null) {
+                    dropBox.addText(TAG_STORAGE_TRIM, scrubPath(path)
                         + " " + bytes + " " + time);
+                }
 
                 final VolumeRecord rec = findRecordForPath(path);
                 if (rec != null) {
-- 
2.11.0

